integration/tests/common/testnet_state_tracker.go

```diff
@@ -104,3 +110,66 @@ func (tst *TestnetStateTracker) Track(t *testing.T, ctx context.Context, ghost *
 		}
 	}()
 }
+
+// WaitUntilFinalizedStateCommitmentChanged waits until a different state commitment for a finalized block is received
```

Comment:
No new logic, just moved from receipt state to here, and made accesses to other state trackers through thread-safe methods.